netbox_cmdb/netbox_cmdb/protect.py

```diff
@@ -0,0 +1,34 @@
+MODELS_LINKED_TO_DEVICE = {}
+MODELS_LINKED_TO_IP_ADDRESS = {}
+
+
+def from_device_name_change(*fields):
+    def decorator(cls):
+        if cls not in MODELS_LINKED_TO_DEVICE:
+            MODELS_LINKED_TO_DEVICE[cls] = set()
+
+        if not fields:
+            return cls
+
+        for field in fields:
+            MODELS_LINKED_TO_DEVICE[cls].add(field)
+
+        return cls
+
+    return decorator
+
+
+def from_ip_address_change(*fields):
```

Comment:
Seems to mostly duplicate the previous function, perhaps it could be factored out, but may not be worth it, your call.

Reply:
I prefer to keep two decorators for UX.
And given we only have two similar functions, I think factorization is overkill and would only bring complexity.

We will consider factorization if we start having more very similar decorator